### DISCUSSION TITLE
How to use setvalue in same create form for updating

### DISCUSSION THREAD

**Discussion by Sharathvasudevan09051991** [react-hook-form/react-hook-form#2398] [Q&A]
For adding its working fine while coming to edit using the same form facing errors. how to use set value for fetching data to input fields and updating forms.

const AddEmployee = ({createEmployee, employee}) => {

const {register, handleSubmit ,setValue,  errors} = useForm({
    mode: 'onBlur',
    reValidateMode: 'onChange'
});

useEffect(() => {
 setValue('empId', 'employee.empId')
}, [setValue, employee]);

const onSubmit = (data) => {
    debugger;
    createEmployee({data})
}

    return (
        <div>
             <div className="row p-2" >
        <form onSubmit={handleSubmit(onSubmit)} autoComplete="off">
          <div className=" p-4">

            <div className="form-group">
              <input
                type="text"
                className="form-control"
                placeholder="Employee ID"
                name="empId"   
                ref={register({required : true})}         
              />
                {errors.empId && <span>This field is required</span>}
            </div>

            <div className="form-group">
              <input
                type="text"
                className="form-control"
                placeholder="Name"
                name="name"
                ref={register({required : true})}         
                />
                  {errors.name && <span>This field is required</span>}
              </div>

            <div className="form-group">
              <input
                type="number"
                className="form-control"
                placeholder="Age"
                name="age"
                ref={register({required : true})}         
                />
                  {errors.age && <span>This field is required</span>}
              </div>

            <div className="form-group">
              <input
                type="email"
                className="form-control"
                placeholder="Email"
                name="email"      
                ref={register({required : true})}         
                />
                  {errors.email && <span>This field is required</span>}
              </div>
            <div className="form-group">
              <input
                type="number"
                className="form-control"
                placeholder="Mobile"
                name="mobile"              
                ref={register({required : true})}         
                />
                  {errors.mobile && <span>This field is required</span>}
              </div>

            <div className="form-group">
              <input
                type="number"
                name="salary"
                className="form-control"
                placeholder="Salary"
                ref={register({required : true})}         
                />
                  {errors.salary && <span>This field is required</span>}
              </div>

            <div className="form-group">
              <input
                type="password"
                name="password"
                className="form-control"
                placeholder="Password"
                ref={register({required : true})}         
                />
                  {errors.password && <span>This field is required</span>}
              </div>

            <div className="form-group">
              <textarea
                rows="2"
                s="50"
                name="address"
                className="form-control"
                placeholder="Address"        
                ref={register({required : true})}         
                />
                  {errors.address && <span>This field is required</span>}
              </div>
            {/* <div className="">
              <input type="radio" id="permanent" name="permanent address"  className="form-control" value="isPermanent"  />
              Permanent address

          </div> */}
            <div className="col mt-2 py-2">
              <input type="submit" className="form-group btn btn-primary px-4" value="Submit" />
            </div>

          </div>
        </form>

    </div>
            
        </div>
    )
}

export default connect(null, {createEmployee})(AddEmployee);

**Comment by bluebill1049** [comment]
hey do you have a simple codesandbox to reproduce this problem?

**Reply by Sharathvasudevan09051991** [reply]
I am not having that .But the question is

Is it possible to use same form for both inserting and updating ?

And please provide any links or documentation to overcome from this issue.

By using getById function in actions. I got that particular object which to edit. While fetching its problem for me.

const upSertEmployee = ({employee}) => {

useEffect(() => {
setValue("empID", employee.empId)
},[employee])
}
Errors: null error in "empId"

**Reply by bluebill1049** [reply]
you can set input value, the error is at your action/fetch

**Reply by Sharathvasudevan09051991** [reply]
Sorry to say this. Iam new to this one. Could you show any examples or codesandbox  link to me.

**Reply by bluebill1049** [reply]
https://react-hook-form.com/api#setValue

**Reply by Sharathvasudevan09051991** [reply]
Yeah i already took a look on this link. I am getting values from action and also that particular employee object in that component. In useeffect i tried to fetch using setValue. But its prompting errors. Am i using correct way to fetching.

**Reply by bluebill1049** [reply]
i don't have much context, but you should try setValue without fetch first, see if there is an error in the first place.
